gtsfm/utils/graph.py

```diff
@@ -32,9 +35,13 @@ def get_nodes_in_largest_connected_component(edges: List[Tuple[int, int]]) -> Li
     input_graph = nx.Graph()
     input_graph.add_edges_from(edges)
 
-    # get the largest connected component
-    largest_cc = max(nx.connected_components(input_graph), key=len)
-    subgraph = input_graph.subgraph(largest_cc).copy()
+    # Log the sizes of the connected components.
+    cc_sizes = [len(x) for x in sorted(list(nx.connected_components(input_graph)))]
+    logger.info("Connected component sizes: %s nodes.", str(cc_sizes))
+
+    # Get the largest connected component.
+    largest_cc_nodes = max(nx.connected_components(input_graph), key=len)
+    subgraph = input_graph.subgraph(largest_cc_nodes).copy()
```

Comment:
could you change this API to return the subgraph for every component? its potentially useful if we decide to optimize multiple connected components. And will also help you save those metrics..